x/supernode/v1/keeper/msg_server_start_supernode.go

```diff
@@ -27,14 +28,31 @@ func (k msgServer) StartSupernode(goCtx context.Context, msg *types2.MsgStartSup
 		return nil, err
 	}
 
-	if len(supernode.States) == 0 || supernode.States[len(supernode.States)-1].State != types2.SuperNodeStateActive {
-		supernode.States = append(supernode.States, &types2.SuperNodeStateRecord{
-			State:  types2.SuperNodeStateActive,
-			Height: ctx.BlockHeight(),
-		})
-	} else {
+	if len(supernode.States) == 0 {
+		return nil, errorsmod.Wrapf(sdkerrors.ErrInvalidRequest, "supernode is in an invalid state")
+	}
+	
+	currentState := supernode.States[len(supernode.States)-1].State
+	
+	// Check if supernode is disabled (terminal state)
+	if currentState == types2.SuperNodeStateDisabled {
+		return nil, errorsmod.Wrapf(sdkerrors.ErrInvalidRequest, "supernode is disabled and must be re-registered")
+	}
+	
+	// Check if already active
+	if currentState == types2.SuperNodeStateActive {
 		return nil, errorsmod.Wrapf(sdkerrors.ErrInvalidRequest, "supernode is already started")
 	}
+	
+	// Can only start from stopped state
+	if currentState != types2.SuperNodeStateStopped {
+		return nil, errorsmod.Wrapf(sdkerrors.ErrInvalidRequest, "supernode must be in stopped state to start")
+	}
+	
+	supernode.States = append(supernode.States, &types2.SuperNodeStateRecord{
+		State:  types2.SuperNodeStateActive,
+		Height: ctx.BlockHeight(),
+	})
```

Comment:
also would be nice to add in EmitEvent the old_state of supernode and height when it was started:
```
sdk.NewAttribute(types2.AttributeKeyOldState, currentState.String()),
sdk.NewAttribute(types2.AttributeKeyHeight, strconv.FormatInt(ctx.BlockHeight(), 10)),
```